libcxx/test/std/algorithms/algorithms.results/out_value_result.pass.cpp

```diff
@@ -0,0 +1,124 @@
+//===----------------------------------------------------------------------===//
```

Comment:
I've started working on this in https://reviews.llvm.org/D121436 (and didn't really work on it because nobody knew what it was for). You should probably make sure that any comments there are addressed here and add any additional tests from there.